### PR TITLE
FIT-381: Add all the codenarc rules for linting/static analysis

### PR DESCRIPTION
**Pull request by edgibbs**
Before adding a build I wanted to get all the code following the same linting/static analysis rules.
There's a lot of changes here, but I tested it with a PR build on dashboard to make sure there
aren't any surprises introduced as there were some changes to production code.

Next up will be a Jenkinsfile and PR/Master build to actually run the linter and unit tests.

### CODE REVIEW

File: src/gov/ca/cwds/jenkins/PullRequestMergedTrigger.groovy [hunk 1]
```diff
@@ -6,13 +6,13 @@ class PullRequestMergedTrigger {
       genericVariables: [
         [key: 'pull_request_action', value: 'action'],
         [key: 'pull_request_merged', value: 'pull_request.merged'],
-        [key: 'pull_request_event', value: 'pull_request']
+        [key: 'pull_request_event', value: 'pull_request'],
```
Comment (mariamcws) [inline src/gov/ca/cwds/jenkins/PullRequestMergedTrigger.groovy]: shouldn't the comma not be there?

File: src/gov/ca/cwds/jenkins/PullRequestMergedTrigger.groovy [hunk 2]
```diff
@@ -6,13 +6,13 @@ class PullRequestMergedTrigger {
...
       ],
       token: "$tokenParameter",
       regexpFilterText: '$pull_request_action:$pull_request_merged',
       regexpFilterExpression: '^closed:true$',
       causeString: 'Triggered by a PR merge',
-      printContributedVariables: false
+      printContributedVariables: false,
```
Comment (mariamcws) [inline src/gov/ca/cwds/jenkins/PullRequestMergedTrigger.groovy]: shouldn't the comma not be there?

File: src/gov/ca/cwds/jenkins/PullRequestMergedTrigger.groovy (hunk 1 above)
Comment (edgibbs) [reply src/gov/ca/cwds/jenkins/PullRequestMergedTrigger.groovy]: So the CodeNarc default setting is here:

http://codenarc.sourceforge.net/codenarc-rules-convention.html

And it defaults under the TrailingComma Rule to be true, so I went with that.  The idea is you can easily refactor the order.  I don't care that much I can flip it to false.

File: src/gov/ca/cwds/jenkins/PullRequestMergedTrigger.groovy (hunk 2 above)
Comment (edgibbs) [reply src/gov/ca/cwds/jenkins/PullRequestMergedTrigger.groovy]: See the conversation above.

File: src/gov/ca/cwds/jenkins/PullRequestMergedTrigger.groovy (hunk 1 above)
Comment (mariamcws) [reply src/gov/ca/cwds/jenkins/PullRequestMergedTrigger.groovy]: ok

**Pull request merged**